crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -3392,13 +3410,29 @@ impl AuthorityPerEpochStore {
                             "Received RandomnessDkgMessage from {:?}",
                             authority.concise()
                         );
-                        match bcs::from_bytes(bytes) {
-                            Ok(message) => randomness_manager.add_message(authority, message)?,
-                            Err(e) => {
-                                warn!(
-                                    "Failed to deserialize RandomnessDkgMessage from {:?}: {e:?}",
-                                    authority.concise(),
-                                );
+                        if self.protocol_config.dkg_version() == 0 {
+                            // old message was not an enum
+                            match bcs::from_bytes(bytes) {
+                                Ok(message) => randomness_manager
+                                    .add_message(authority, VersionedDkgMessage::V0(message))?,
+                                Err(e) => {
```

Comment:
a lot of this duplicated code could be factored out, right? since the only thing that's different is the deserialization call, but the error handling and the call to `add_message` are the same